### PR TITLE
Design Picker: Replace generated design thumbnails with iframe

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

* This PR is focusing on the POC of using the `<iframe />` as the generated design thumbnail to see whether the resolution and performance are acceptable or not. Below are the demonstrations with/without sandboxed API.

**Comparison**

| | iframe | mShots without cache | mShots with cache |
| - | - | - | - |
| Resolution | Same | Same | Same |
| Performance | Faster | Slower | Faster |

As you see, if we use the `<iframe />` as the thumbnail, the resolution looks the same but the performance is better than the `mShots` without cache (the first one). When the `mShots` has a cache, the performance might be similar. Hence, I think it's good to replace the `mShots` with the iframe preview Any thoughts?

**Without sandboxed API**

| iframe | mShots |
| - | - |
| <video src="https://user-images.githubusercontent.com/13596067/175236151-86c45f1a-fc15-4828-84af-d4c4394db813.mov" /> | <video src="https://user-images.githubusercontent.com/13596067/175236188-2f5dba90-a5db-4f10-8c62-01ba93577d74.mov" /> |

**With Sandboxed API**

| iframe | mShots |
| - | - |
| <video src="https://user-images.githubusercontent.com/13596067/175236356-b850738f-f0f5-4d2b-a1d4-a00196bae830.mov" /> | <video src="https://user-images.githubusercontent.com/13596067/175236325-22e64c41-6b81-4cc0-bc1b-45f07f5afa12.mov" /> |

#### Testing Instructions

* Go to `/setup?siteSlug=<your_site>`
* Select "Promote myself or business" goal
* Select a vertical
* When you land on the Design Picker, the thumbnails have to look the same as the mShots image

Related to 68-gh-Automattic/ganon-issues

### CODE REVIEW

**Comment by taipeicoder** [comment]
My two cents: One of the major benefits which we chose to use mShots was because of caching. As we add more verticalization features, I feel that caching will be harder and harder to do, so I think moving away from mShots is a good idea.

That being said, being this close to the feature launch, I would prefer to move this post-launch. LMK what you all think!

**Comment by arthur791004** [comment]
> That being said, being this close to the feature launch, I would prefer to move this post-launch. LMK what you all think!

Yup, I agree with that. So, I set the milestone to `v13n evolution (post-MVP)` and this one is just a demo

**Comment by miksansegundo** [comment]
Something that we have to consider is that on today's Onboarding & Verticals Cross-Discipline Sync, we have been told that the future plan for the design picker is increasing dramatically the size and the number of the recipe thumbnails listed. It is still a work in progress. See the image to get an idea.

Verticals Cross-Discipline Organization

**Comment by taipeicoder** [comment]
After giving the PoC a whirl, here are my thoughts on using iframes:

Performance wise, I think using iframes is acceptable, using sandboxed API the thumbnails render in ~5 seconds, which is better than mShots with cold cache, but obviously not better than cached mShots. Using production API, the thumbnails render in ~1 second for me, which is comparable to mShots with cache.

Implementation wise, not very comfortable using `<div>` within `<span>` 😅 but this is trivial to improve.

Resource wise, iframe uses significantly a lot more resources than mShots. See comparison:
| iframe | mShots |
| ------ | -------- |
| Screen Shot 2022-06-24 at 4 21 00 PM |  Screen Shot 2022-06-24 at 4 20 34 PM |

This might cause slowness for mobile users that don't have good Internet speed. And it wouldn't scale well if we decide to show more than three thumbnails. For instance, in the design from the comment above, we see about 15 thumbnails.

In conclusion, IMO with the current design of showing only three thumbnails, I think it makes sense to switch to iframe since my experience is that they loaded in ~1 seconds using production API. I would like to know if others experience similar speed.

**Comment by taipeicoder** [comment]
@arthur791004 one a11y issue I encountered was that while the focus is on the thumbnail, tabbing will change focus to the embedded page instead of the next thumbnail 🥲.

**Comment by arthur791004** [comment]
> @arthur791004 one a11y issue I encountered was that while the focus is on the thumbnail, tabbing will change focus to the embedded page instead of the next thumbnail 🥲.

Nice catch! Fixed!

**Review by taipeicoder — APPROVED** [review]
LGTM, let's 🚢 !
